RealmTasks Apple/RealmTasks iOS/CellPresenter.swift

```diff
@@ -115,17 +114,12 @@ class CellPresenter<Item: Object where Item: CellPresentable> {
         }
 
         let item = editingCell.item
-        guard !(item as Object).invalidated else {
-            tableView.reloadData()
-            return
-        }
         if item.text.isEmpty {
-            try! item.realm?.write {
-                item.realm!.delete(item)
-            }
+            let indexPath = NSIndexPath(forRow: items.indexOf(item)!, inSection: 0)
+            item.realm!.delete(item)
+            viewController.tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Top)
         }
-
-        viewController.didUpdateList()
+        viewController.finishUIWrite()
```

Comment:
the `finishUIWrite()` method is how writes 1) done on the UI thread; and 2) already applied to the UI; should be committed.

Transaction was started with `items.realm?.beginWrite()` because there's no `beginUIWrite()` method because there's no distinction between UI writes vs background writes when beginning the transaction, only when committing it.

I could add a `beginUIWrite()` method on `ViewControllerProtocol` for symmetry with `finishUIWrite()`, but it would just be for symmetry's sake.